src/coreclr/jit/compiler.h

```diff
@@ -994,13 +994,14 @@ class LclVarDsc
         regMaskTP regMask = RBM_NONE;
         if (GetRegNum() != REG_STK)
         {
-            if (varTypeUsesFloatReg(TypeGet()))
+            if (varTypeUsesIntReg(this))
             {
-                regMask = genRegMaskFloat(GetRegNum(), TypeGet());
+                regMask = genRegMask(GetRegNum());
             }
             else
             {
-                regMask = genRegMask(GetRegNum());
+                assert(varTypeUsesFloatReg(this));
+                regMask = genRegMaskFloat(GetRegNum(), TypeGet());
```

Comment:
```suggestion
                regMask = genRegMaskFloat(GetRegNum() ARM_ARG(TypeGet()));
```

We should update the definition to:

```c++
 inline regMaskTP genRegMaskFloat(regNumber reg ARM_ARG(var_types type = TYP_DOUBLE));
```

Reply:
Fixed.